tests/e2e/__testers__.ts

```diff
@@ -92,60 +201,64 @@ export const testTemplate = (params: {
 
-    const prisma = await ctx.getPrisma()
-    const prismaAdmin = await ctx.getPrismaAdmin()
+    // TODO Test the Vercel API (next dev for Blog template)
+    // await ctx.fs.writeAsync('.vercel/project.json', {
+    //   projectId: 'prj_6yrTe9CGQagAQwGjr7JEejkxhz3A',
+    //   orgId: 'team_ASKXQ5Yc1an2RqJc5BCI9rGw',
+    // })
+  })
 
-    try {
-      /**
-       * Test 2
-       * Check that the template migration script works.
-       */
+  /**
+   * Test 2
+   * Check that the template migration script works.
+   */
+  if (params.templateName !== 'Empty') {
+    it(`${params.templateName} - template migration script should work`, async () => {
       await ctx.dropTestDatabase()
-      await prismaAdmin.$executeRawUnsafe(`create database ${ctx.databaseName}`)
+      await ctx.createTestDatabase()
+
+      console.log('Get getPrisma()')
+      const prisma = await ctx.getApplicationPrisma()
       await Reflector.Client.runMigrationScript(
         prisma,
         ctx.template.migrationScript,
-        // TODO test a matrix of data sources
-        'postgres'
+        params.datasourceProvider
       )
-
-      /**
-       * Test 3
-       * Check the seed again but this time using the derived seed function.
-       */
+    })
+  }
+  /**
+   * Test 3
+   * Check the seed again but this time using the derived seed function.
+   */
+  if (params.templateName !== 'Empty') {
+    it.skip(`${params.templateName} - seed using the derived seed function should work`, async () => {
+      const prisma = await ctx.getApplicationPrisma()
       // TODO improve seed scripts to return reports that we can use to capture feedback here not to mention for users generally.
       await ctx.template.seed({ prisma })
-    } finally {
-      await Promise.all([prisma.$disconnect(), prismaAdmin.$disconnect()])
-    }
+    })
+  }
 
-    /**
-     * Test 4
-     * Check the development project script. For most templates this will run some kind of sandbox script against the database.
-     *
-     * The Nextjs template launches next dev for its dev script and thus is exempt from this test.
-     */
+  /**
+   * Test 4
+   * Check the development project script. For most templates this will run some kind of sandbox script against the database.
+   *
+   * The Nextjs template launches next dev for its dev script and thus is exempt from this test.
+   */
+  it(`${params.templateName} - development project script should work`, async () => {
     if (ctx.template._tag !== 'Nextjs') {
       const devResult = ctx.run(`npm run dev`, { reject: true })
       expect(devResult.stderr).toMatch('')
       expect(devResult.stdout).toMatch(params.expectedDevOutput)
     }
-
-    // TODO Test the Vercel API (next dev for Blog template)
-    // await ctx.fs.writeAsync('.vercel/project.json', {
-    //   projectId: 'prj_6yrTe9CGQagAQwGjr7JEejkxhz3A',
-    //   orgId: 'team_ASKXQ5Yc1an2RqJc5BCI9rGw',
-    // })
```

Comment:
This TODO is still relevant.

Reply:
It is still in the main test. It looks removed because there are new tests now.
I moved it down to the last test.